log/logClient.go

```diff
@@ -202,124 +210,143 @@ func regexMatcher(filter *regexp.Regexp, res string) bool {
-			str = str + fmt.Sprintf("Cluster Name: %s\n", res.ClusterName)
-			str = str + fmt.Sprintf("Host Name: %s\n", res.HostName)
+		if res.NamespaceName != "" {
+			str = str + fmt.Sprintf("Namespace Name: %s\n", res.NamespaceName)
+			str = str + fmt.Sprintf("Pod Name: %s\n", res.PodName)
+			str = str + fmt.Sprintf("Container ID: %s\n", res.ContainerID)
+			str = str + fmt.Sprintf("Container Name: %s\n", res.ContainerName)
+		}
 
-			if res.NamespaceName != "" {
-				str = str + fmt.Sprintf("Namespace Name: %s\n", res.NamespaceName)
-				str = str + fmt.Sprintf("Pod Name: %s\n", res.PodName)
-				str = str + fmt.Sprintf("Container ID: %s\n", res.ContainerID)
-				str = str + fmt.Sprintf("Container Name: %s\n", res.ContainerName)
-			}
+		if len(res.PolicyName) > 0 {
+			str = str + fmt.Sprintf("Policy Name: %s\n", res.PolicyName)
+		}
 
-			if len(res.PolicyName) > 0 {
-				str = str + fmt.Sprintf("Policy Name: %s\n", res.PolicyName)
-			}
+		if len(res.Severity) > 0 {
+			str = str + fmt.Sprintf("Severity: %s\n", res.Severity)
+		}
 
-			if len(res.Severity) > 0 {
-				str = str + fmt.Sprintf("Severity: %s\n", res.Severity)
-			}
+		if len(res.Tags) > 0 {
+			str = str + fmt.Sprintf("Tags: %s\n", res.Tags)
+		}
 
-			if len(res.Tags) > 0 {
-				str = str + fmt.Sprintf("Tags: %s\n", res.Tags)
-			}
+		if len(res.Message) > 0 {
+			str = str + fmt.Sprintf("Message: %s\n", res.Message)
+		}
 
-			if len(res.Message) > 0 {
-				str = str + fmt.Sprintf("Message: %s\n", res.Message)
-			}
+		str = str + fmt.Sprintf("Type: %s\n", res.Type)
+		str = str + fmt.Sprintf("Source: %s\n", res.Source)
+		str = str + fmt.Sprintf("Operation: %s\n", res.Operation)
+		str = str + fmt.Sprintf("Resource: %s\n", res.Resource)
 
-			str = str + fmt.Sprintf("Type: %s\n", res.Type)
-			str = str + fmt.Sprintf("Source: %s\n", res.Source)
-			str = str + fmt.Sprintf("Operation: %s\n", res.Operation)
-			str = str + fmt.Sprintf("Resource: %s\n", res.Resource)
+		if len(res.Data) > 0 {
+			str = str + fmt.Sprintf("Data: %s\n", res.Data)
+		}
 
-			if len(res.Data) > 0 {
-				str = str + fmt.Sprintf("Data: %s\n", res.Data)
-			}
+		if len(res.Action) > 0 {
+			str = str + fmt.Sprintf("Action: %s\n", res.Action)
+		}
 
-			if len(res.Action) > 0 {
-				str = str + fmt.Sprintf("Action: %s\n", res.Action)
+		str = str + fmt.Sprintf("Result: %s\n", res.Result)
+	}
+
+	if o.LogPath == "stdout" {
+		fmt.Printf("%s", str)
+	} else {
+		StrToFile(str, o.LogPath)
+	}
+	return nil
+}
+
+// WatchAlerts Function
+func (fd *Feeder) WatchAlerts(o Options) error {
+	fd.WgClient.Add(1)
+	defer fd.WgClient.Done()
+
+	if o.Limit > 0 {
+		for i = 0; i < o.Limit; i++ {
+			res, err := fd.alertStream.Recv()
+			if err != nil {
+				fmt.Printf("Failed to receive an alert (%s)\n", err.Error())
+				break
 			}
+			_ = watchAlertsHelper(res, o)
 
-			str = str + fmt.Sprintf("Result: %s\n", res.Result)
 		}
+		Limitchan <- true
+
+	} else {
+		for fd.Running {
+			res, err := fd.alertStream.Recv()
```

Comment:
Is the possible to remove this code duplication ... I mean `Recv() ... watchAlertsHelper()`

Reply:
I think currently only code duplication section is this one 
```
res, err := fd.logStream.Recv()
if err != nil {
	fmt.Printf("Failed to receive an alert (%s)\n", err.Error())
	break
}
_ = WatchLogsHelper(res, o)
```

I tried to add it in another function ... but it didn't help